ext/simdjson/src/bindings.cpp

```diff
@@ -0,0 +1,238 @@
+simdjson::dom::parser parser;
+
+WARN_UNUSED
+simdjson::error_code
+build_parsed_json_cust(
+	simdjson::dom::element &doc, const char *buf, size_t len, bool realloc_if_needed, u_short depth = simdjson::DEFAULT_MAX_DEPTH
+) {
+    auto error = parser.allocate(len, depth);
+
+    if (error) {
+        return error;
+    }
+
+    error = parser.parse(buf, len, realloc_if_needed).get(doc);
+    if (error) {
+        return error;
+    }
+
+    return simdjson::SUCCESS;
+}
+
+static zval create_array(simdjson::dom::element element)
+{
+    zval v;
+
+    switch (element.type()) {
+        //ASCII sort
+        case simdjson::dom::element_type::STRING :
+            ZVAL_STRING(&v, std::string_view(element).data());
+            break;
+        case simdjson::dom::element_type::INT64 : ZVAL_LONG(&v, int64_t(element));
+            break;
+        case simdjson::dom::element_type::UINT64 : ZVAL_LONG(&v, uint64_t(element));
+            break;
+        case simdjson::dom::element_type::DOUBLE : ZVAL_DOUBLE(&v, double(element));
+            break;
+        case simdjson::dom::element_type::BOOL :
+            ZVAL_BOOL(&v, bool(element));
+            break;
+        case simdjson::dom::element_type::NULL_VALUE :
+            ZVAL_NULL(&v);
+            break;
+        case simdjson::dom::element_type::ARRAY :
+            zval arr;
+            array_init(&arr);
+
+            for (simdjson::dom::element child : simdjson::dom::array(element)) {
+                zval value = create_array(child);
+                add_next_index_zval(&arr, &value);
+            }
+
+            v = arr;
+            break;
+        case simdjson::dom::element_type::OBJECT :
+            zval obj;
+            array_init(&obj);
+
+            for (simdjson::dom::key_value_pair field : simdjson::dom::object(element)) {
+                zval value = create_array(field.value);
+                add_assoc_zval_ex(&obj, field.key.data(), strlen(field.key.data()), &value);
+            }
+            v = obj;
+            break;
+        default:
+            break;
+    }
+
+    return v;
+}
+
+static zval create_object(simdjson::dom::element element)
+{
+    zval v;
+
+    switch (element.type()) {
+        //ASCII sort
+        case simdjson::dom::element_type::STRING :
+            ZVAL_STRING(&v, std::string_view(element).data());
+            break;
+        case simdjson::dom::element_type::INT64 : ZVAL_LONG(&v, int64_t(element));
+            break;
+        case simdjson::dom::element_type::UINT64 : ZVAL_LONG(&v, uint64_t(element));
+            break;
+        case simdjson::dom::element_type::DOUBLE : ZVAL_DOUBLE(&v, double(element));
+            break;
+        case simdjson::dom::element_type::BOOL :
+            ZVAL_BOOL(&v, bool(element));
+            break;
+        case simdjson::dom::element_type::NULL_VALUE :
+            ZVAL_NULL(&v);
+            break;
+        case simdjson::dom::element_type::ARRAY :
+            zval arr;
```

Comment:
nit: It seems strange to create a different zval and then copy it by value into `v`. Can this just `array_init(&v);` instead (same for other switch cases and other function)